api/clients/v2/verification/cert_verifier.go

```diff
@@ -219,105 +168,82 @@ func (cv *CertVerifier) GetNonSignerStakesAndSignature(
-		return nil, fmt.Errorf("get verifier caller: %w", err)
+		return nil, fmt.Errorf("fetch latest block number: %w", err)
+	}
+
+	certVerifierAddress, err := cv.certVerifierAddressProvider.GetCertVerifierAddress(blockNumber)
+	if err != nil {
+		return nil, fmt.Errorf("get cert verifier address: %w", err)
+	}
+
+	// if the quorum numbers for the active cert verifier address have already been cached, return them immediately
+	cachedQuorumNumbers, ok := cv.requiredQuorums.Load(certVerifierAddress)
+	if ok {
+		castQuorums, ok := cachedQuorumNumbers.([]uint8)
+		if !ok {
+			return nil, fmt.Errorf("expected quorum numbers to be []uint8")
+		}
+		return castQuorums, nil
+	}
+
+	// quorum numbers weren't cached, so proceed to fetch them
+	certVerifierCaller, err := cv.getVerifierCallerFromAddress(certVerifierAddress)
+	if err != nil {
+		return nil, fmt.Errorf("get verifier caller from address: %w", err)
 	}
 
 	quorumNumbersRequired, err := certVerifierCaller.QuorumNumbersRequiredV2(&bind.CallOpts{Context: ctx})
 	if err != nil {
 		return nil, fmt.Errorf("get quorum numbers required: %w", err)
 	}
 
+	cv.requiredQuorums.Store(certVerifierAddress, quorumNumbersRequired)
+
 	return quorumNumbersRequired, nil
 }
 
-// MaybeWaitForBlockNumber waits until the internal eth client has advanced to a certain targetBlockNumber, unless
-// configured pollInterval is <= 0, in which case this method will NOT wait for the internal client to advance.
+// getVerifierCallerFromBlockNumber returns a ContractEigenDACertVerifierCaller that corresponds to the input block
+// number
 //
-// This method will check the current block number of the internal client every CertVerifier.pollInterval duration.
-// It will return nil if the internal client advances to (or past) the targetBlockNumber. It will return an error
-// if the input context times out, or if any error occurs when checking the block number of the internal client.
+// If the eth node hasn't yet advanced to the input block number, this method will wait until that block is reached
+// before attempting to get the verifier caller.
 //
-// This method is synchronized in a way that, if called by multiple goroutines, only a single goroutine will actually
-// poll the internal eth client for most recent block number. The goroutine responsible for polling at a given time
-// updates an atomic integer, so that all goroutines may check the most recent block without duplicating work.
-func (cv *CertVerifier) MaybeWaitForBlockNumber(ctx context.Context, targetBlockNumber uint64) error {
-	if cv.pollInterval <= 0 {
-		// don't wait for the internal client to advance
-		return nil
-	}
-
-	if cv.latestBlockNumber.Load() >= targetBlockNumber {
-		// immediately return if the local client isn't behind the target block number
-		return nil
+// This method caches ContractEigenDACertVerifierCaller instances, since their construction requires acquiring a lock
+// and parsing json, and is therefore not trivially inexpensive.
```

Comment:
nit, `non-trivially`?

Reply:
Changed to "non-trivially expensive" (which I believe carries the same meaning of "not trivially inexpensive", but is easier to parse)